tidy3d/components/eme/simulation.py

```diff
@@ -542,9 +553,34 @@ def _post_init_validators(self) -> None:
         self._validate_modes_size()
         self._validate_sweep_spec()
         self._validate_symmetry()
+        self._validate_constraint()
         # self._warn_monitor_interval()
         log.end_capture(self)
 
+    def _validate_constraint(self):
```

Comment:
curious why this needs to be a post init validator not a regular one?

Reply:
It needs the `sim.eme_grid` to already be initialized, in order to check the max number of modes